lib/bumblebee/text/question_answering.ex

```diff
@@ -0,0 +1,176 @@
+defmodule Bumblebee.Text.QuestionAnswering do
+  alias Bumblebee.Tokenizer
+  alias Bumblebee.Shared
+  alias Bumblebee.Utils
+  alias Axon
+
+  @moduledoc false
+
+  def question_answering(model_info, tokenizer, opts \\ []) do
+    %{model: model, params: params, spec: spec} = model_info
+    Shared.validate_architecture!(spec, :for_question_answering)
+
+    opts =
+      Keyword.validate!(opts, [
+        :compile,
+        doc_stride: 128,
+        top_k: 1,
+        defn_options: []
+      ])
+
+    top_k = opts[:top_k]
+    compile = opts[:compile]
+    defn_options = opts[:defn_options]
+    doc_stride = opts[:doc_stride]
+
+    batch_size = compile[:batch_size]
+    sequence_length = compile[:sequence_length]
+
+
+    if compile != nil and (batch_size == nil or sequence_length == nil) do
+      raise ArgumentError,
+            "expected :compile to be a keyword list specifying :batch_size and :sequence_length, got: #{inspect(compile)}"
+    end
+
+    {_init_fun, predict_fun} = Axon.build(model)
+
+    scores_fun = fn params, input ->
+      # input = Utils.Nx.composite_flatten_batch(input)
+      output = predict_fun.(params, input)
+    end
+
+    Nx.Serving.new(
+      fn ->
+        predict_fun =
+          Shared.compile_or_jit(scores_fun, defn_options, compile != nil, fn ->
+            inputs = %{
+              "input_ids" => Nx.template({batch_size, sequence_length}, :s64),
+              "attention_mask" => Nx.template({batch_size, sequence_length}, :s64)
+            }
+
+            [params, inputs]
+          end)
+
+        fn inputs ->
+          inputs = Shared.maybe_pad(inputs, batch_size)
+
+          scores = predict_fun.(params, inputs)
+        end
+      end,
+      batch_size: batch_size
+    )
+    |> Nx.Serving.client_preprocessing(fn input ->
+      {inputs, multi?} =
+        Shared.validate_serving_input!(input, fn
+          %{question: question, context: context}
+          when is_binary(question) and is_binary(context) ->
+            {:ok, {question, context}}
+
+          other ->
+            {:error,
+             "expected input map with :question and :context keys, got: #{inspect(other)}"}
+        end)
+
+      inputs =
+        Bumblebee.apply_tokenizer(tokenizer, inputs,
+          length: sequence_length,
+          return_special_tokens_mask: true,
+          return_offsets: true
+        )
+
+      {Nx.Batch.concatenate([inputs]), {inputs, multi?}}
```

Comment:
```suggestion
      all_inputs =
        Bumblebee.apply_tokenizer(tokenizer, inputs,
          length: sequence_length,
          return_special_tokens_mask: true,
          return_offsets: true
        )

      inputs = Map.take(all_inputs, ["input_ids", "attention_mask"])

      {Nx.Batch.concatenate([inputs]), {all_inputs, multi?}}
```

that's likely what the error is coming from, the tokenizer gives us `start_offsets` and `end_offsets`, and we want to use this in postprocessing, but we don't want to pass it into the model, since we compile it to only expect `input_ids` and `attentino_mask`.

Reply:
Ahh Im close, but now im getting tensors as the end and start, do I need to normalize?

Reply:
`Nx.to_number` should be enough I think.

Reply:
Hmm Im getting this 
```
left:  %{
              predictions: [
                %{
                  text: "blue",
                  start: answer_start_index,
                  end: answer_end_index,
                  score: score
                } | rest
              ]
            }
     right: %{
              results: [%{end: #Nx.Tensor<
      s64[32]
      [0, 4, 7, 12, 17, 22, 23, 0, 0, 2, 3, 8, 13, 16, 21, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0]
    >, score: 0, start: #Nx.Tensor<
      s64[32]
      [0, 0, 5, 8, 13, 18, 23, 0, 0, 0, 3, 4, 9, 14, 17, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0]
    >, text: " "}]
            }
```